File: ios/ReactToNative/LoginViewManager/LoginViewEventManager.m
//
//  LoginViewEventManager.m
//  ReactToNative
//
//  Created by Pierluigi Codella on 16/12/24.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LoginViewEventManager, NSObject)

RCT_EXTERN_METHOD(onLoginSuccess: (NSDictionary)event)
RCT_EXTERN_METHOD(onLogoutSuccess)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}


@end

File: ios/ReactToNative/LoginViewManager/LoginViewEventManager.swift
//
//  LoginViewEventManager.swift
//  ReactToNative
//
//  Created by Pierluigi Codella on 16/12/24.
//

import Foundation

@objc(LoginViewEventManager)
public class LoginViewEventManager: NSObject {
    
    
    @objc(onLoginSuccess:)
    func onLoginSuccess(event: NSDictionary) {
        UserManager.shared.isLoggedIn = true
    }
    
    @objc(onLogoutSuccess)
    func onLogoutSuccess() {
        UserManager.shared.isLoggedIn = false
    }
    
}
